table/block_builder.cc

```diff
@@ -60,6 +75,11 @@ void BlockBuilder::Reset() {
   counter_ = 0;
   finished_ = false;
   last_key_.clear();
+  if (data_block_hash_index_builder_) {
+    // use the num_keys_ of current block as an estimate for the next block.
+    data_block_hash_index_builder_->Reset(num_keys_);
```

Comment:
Setting number of buckets of the next block in the Reset() is not intuitive to me.
Another question is that why do you need to preset number of buckets? Can you keep all the <hash, offset_index> pair, and determine the size of the hash table size in the end? In that case we don't need to rely on previous block's number of buckets.